youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,196 @@
+
+    _TESTS = [
+        {
+            "url": "http://www.youmaker.com/v/71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+            "info_dict": {
+                "id": "71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+                "ext": "mp4",
+                "title": "Как сшить шапочку из трикотажа. Плоский шов двойной иглой.",
+                "description": r"re:(?s)^Привет друзья!\n\nВ этом видео я .* представлена www\.iksonmusic\.com$",
+                "thumbnail": r"re:^https?://.*\.(?:jpg|png)$",
+                "duration": 358,
+                "upload_date": "20190614",
+                "uploader": "user_318f21e00e1f8a6b414f20a654d0f4fc7d2053bc",
+                "timestamp": 1560554895,
+                "channel": "Sewing Ideas",
+                "channel_id": "40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "channel_url": r"re:https?://www.youmaker.com/channel/40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "tags": [
+                    "как сшить детскую шапочку из трикотажа",
+                    "как шить двойной иглой трикотаж",
+                ],
+                "categories": ["Life", "How-to & DIY"],
+            },
+            "params": {
+                "skip_download": True,
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+        self.__video_id = None
+        self.__cache = {}
+
+    def _set_id_and_protocol(self, url):
+        self.__video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _api(self, path, cache=False, **kwargs):
+        """
+        call the YouMaker JSON API and return the data
+
+        path:       API endpoint
+        **kwargs:   parameters passed to _download_json()
+        """
+        key = hash((path, compat_urllib_parse_urlencode(kwargs.get("query", {}))))
+        if cache and key in self.__cache:
+            return self.__cache[key]
+
+        url = "%s/v1/api/%s" % (self._base_url, path)
+        info = self._download_json(url, self.__video_id, **kwargs)
+        status = info.get("status", "something went wrong")
+        data = info.get("data")
+        if status != "ok" or data is None:
+            raise ExtractorError(status, expected=True)
+
+        if cache:
+            self.__cache[key] = data
+
+        return data
+
+    def _categories_by_id(self, cid):
```

Comment:
When this is called, `cid` might be `None`, so should the method return immediately in that case?